OpenRA.Mods.Common/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -57,25 +58,39 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 			CPos endPos;
 			WAngle spawnFacing;
 
+			var bounds = map.Bounds;
+			var diagonal = Exts.ISqrt(bounds.Height * bounds.Height + bounds.Width * bounds.Width);
+
 			if (info.BaselineSpawn && mpStart != null)
 			{
 				var spawn = mpStart.Start[owner];
-				var bounds = map.Bounds;
 				var center = new MPos(bounds.Left + bounds.Width / 2, bounds.Top + bounds.Height / 2).ToCPos(map);
 				var spawnVec = spawn - center;
-				startPos = spawn + spawnVec * (Exts.ISqrt((bounds.Height * bounds.Height + bounds.Width * bounds.Width) / (4 * spawnVec.LengthSquared)));
+				startPos = spawn + spawnVec * diagonal / (2 * spawnVec.Length);
 				endPos = startPos;
-				var spawnDirection = new WVec((self.Location - startPos).X, (self.Location - startPos).Y, 0);
-				spawnFacing = spawnDirection.Yaw;
+				spawnFacing = map.FacingBetween(startPos, self.Location, WAngle.Zero);
 			}
 			else
 			{
-				// Start a fixed distance away: the width of the map.
-				// This makes the production timing independent of spawnpoint
-				var loc = self.Location.ToMPos(map);
-				startPos = new MPos(loc.U + map.Bounds.Width, loc.V).ToCPos(map);
-				endPos = new MPos(map.Bounds.Left, loc.V).ToCPos(map);
+				// Start a fixed distance away: the longest straight line across the
+				// map at a given angle. This makes the production timing independent
+				// of spawnpoint.
 				spawnFacing = info.Facing;
+				var diagonalAngle = WAngle.ArcTan(bounds.Width, bounds.Height);
```

Comment:
This looks a lot more complicated than I would expect. Can't we just do `var longestApproach = Math.Max(bounds.Height * 1024 / spawnFacing.Cos(), bounds.Width * 1024 / spawnFacing.Sin());`?

Reply:
No, that would be nonsense. That would always result in a length that is much longer than even the diagonal, except when the angle is exactly along the diagonal.

Reply:
Ok. Reading back my own comment, I realized that that would mean that `Math.Min` might actually work there :sweat_smile: